### PR TITLE
Sabotage dialog

### PR DESCRIPTION
**Pull request by grxver7**

### CODE REVIEW

**Comment by ubeF** [comment]
- The Description of the sabotage-events is dated ("Eigentor" is now only on 11 for example)
- Why does sabotage have it's own dice result instead of re-using the normal dice?
- It would probably be better to use the player-icons on the buttons instead of "a, b, c, d" so it's easier to pick the correct player

**Comment by grxver7** [comment]
Player selection: Player icons are inserted. 

Dice result: The scene (is from the naming convention not so good probably) is intended as information for the player. There should be a label, on which sabotage will be done. For example "You get a sabotage-free card".

**Comment by grxver7** [comment]
Dice results have been adjusted. So 1-6 duel...

**Comment by ubeF** [comment]
Window displaying the different options is displayed behind map

**Comment by ubeF** [comment]
Following players should not be displayed in the "select_duelist" window:
- The player picking his opponent
- Players, which have the attribute "sabotage_card" set to true

Players will be accessible with State.players once my PR goes trough

**Comment by ubeF** [comment]
"select_duelist" scene is for some reason invisible when instantiated

**Comment by ubeF** [comment]
Basically completed. only waiting for merging of:
- [x] #50 Target Selector
- [x] #54 Duels

File: sabotage_field/sabotage_dice_result.gd [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+		"text": "Duell Einsatz: 50 Riesen",
+		"action": func():
+			var target = await get_target()
+			var result = await run_duel(target)
+			# match case over result
+	}
+
+
+	if (dice_value==7 || dice_value==8):
+		result="Big Baller Duell Einsatz: halbes Vermögen"
```
Comment (hfxbse) [inline sabotage_field/sabotage_dice_result.gd]: The variable `result` is not declared in the current scope, seems to be a recent change that broke it.

File: sabotage_field/sabotage_dice_result.gd [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+		action = steal_turn
+	if (dice_value==10):
+		result="Erhalt Sabotage Freikarte"
+		action = sabotage_free_card
+	if (dice_value==11):
+		result="Eigentor Verlust: 50 Riesen"
+		action = lose_50
+	if (dice_value==12):
+		result="Duell Würfelgeld \nWert:gewürfelte Anzahl Riesen*10"
+		action = duel_roll
```
Comment (hfxbse) [inline sabotage_field/sabotage_dice_result.gd]: Mapping the values should also be done with a Dictionary but at least use `elif` for consecutive statements.

File: player/player.gd [hunk 3]
```diff
@@ -19,6 +20,8 @@ var riesen: int:
 	set(amount):
 		if riesen != amount:
 			riesen = amount
+			if riesen < 0:
+				riesen = 0
```
Comment (hfxbse) [inline player/player.gd]: cloud be one line with `riesen = max(0, amount)`.

**Review by hfxbse — CHANGES_REQUESTED** [review]
Looks fine mostly so far.

File: sabotage_field/sabotage_dice_result.gd [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+
+func apply_duel_outcome(attacker, target, bet_amount, result):
+	if result == Duel.DuelEndings.ATTACKER_LOSES:
+		bet_amount *= -1
+	elif result == Duel.DuelEndings.DRAW:
+		bet_amount *= 0
+	
+	attacker += bet_amount
+	target -= bet_amount
```
Comment (hfxbse) [inline sabotage_field/sabotage_dice_result.gd]: This is not working. You probably meant
```gdscript
attacker.riesen += bet_amount
target.riesen -= bet_amount
```

File: sabotage_field/sabotage_dice_result.gd
Comment (hfxbse) [inline sabotage_field/sabotage_dice_result.gd:1]: I would love to see the display options separated from the function as I did for the action dialog.

File: sabotage_field/sabotage_dice_result.gd [hunk 5]
```diff
@@ -0,0 +1,109 @@
+extends CanvasLayer
+
+@onready var label=$CanvasLayer/Panel/Label
```
Comment (hfxbse) [inline sabotage_field/sabotage_dice_result.gd]: The spaces between the equal sign are missing. It should be:
```gdscript
@onready var label = $CanvasLayer/Panel/Label
```

File: project.godot [hunk 6]
```diff
@@ -13,6 +13,7 @@ config_version=5
 config/name="mparty"
 run/main_scene="res://main/main.tscn"
 config/features=PackedStringArray("4.0", "GL Compatibility")
+config/icon="res://player/icon.svg"
```
Comment (hfxbse) [inline project.godot]: This icon is not present anymore, please revert the project file to the one in main.

File: player/player.gd [hunk 7]
```diff
@@ -20,6 +21,7 @@ var riesen: int:
 	set(amount):
 		if riesen != amount:
 			riesen = amount
+			riesen = max(0, riesen)
```
Comment (hfxbse) [inline player/player.gd]: Write it in one line instead.
```gdscript
riesen = max(0, amount)
```

**Review by hfxbse — CHANGES_REQUESTED** [review]
The directory `sabotage_field` should be called `sabotage_dialog` to not create confusion with the actual field for the map.

**Review by hfxbse — APPROVED** [review]
I cannot be border to test it by getting lucky and landing on the matching field, but I do not see why it should not work now haha.

**Pull request merged**